### ISSUE TITLE
Intentional removal of fragmentShaderBarycentric?

### ISSUE THREAD

**Issue by ds-saschasc** [KhronosGroup/MoltenVK#1831]
Tested on macOS 11.7.2 (20G1020) and 11.7.3 (20G1116), MacBook Pro (16-inch, 2019), AMD Radeon Pro 5500M 4 GB

With Vulkan SDK 1.3.216 I was able to use the fragmentShaderBarycentric feature. It seems that this has been removed with Vulkan SDK 1.3.236. Why?

```
VkPhysicalDeviceFragmentShaderBarycentricFeaturesKHR:
-----------------------------------------------------
	fragmentShaderBarycentric = true
```

[vulkaninfo 1.3.216.txt](https://github.com/KhronosGroup/MoltenVK/files/10520800/vulkaninfo.1.3.216.txt)
[vulkaninfo 1.3.236.txt](https://github.com/KhronosGroup/MoltenVK/files/10520801/vulkaninfo.1.3.236.txt)

See also here:
https://vulkan.gpuinfo.org/listdevicescoverage.php?extensionname=VK_KHR_fragment_shader_barycentric&extensionfeature=fragmentShaderBarycentric&platform=all

**Comment by saschasc** [comment]
Could this be the culprit?

https://github.com/KhronosGroup/MoltenVK/commit/b54c7013939483f146080634ab3fcbbc5057e0ea

1.3.216 was released in June 2022. The changes are from July. I would have to test again, but one colleague with another setting didn't get it to work on his Intel GPU (but he only tested, 1.3.224 and 1.3.236 until now.

We were using the barycentric coordinate feature for wireframe rendering and it worked perfectly with 1.3.216.

**Comment by saschasc** [comment]
Strange. I don't know a lot about metal, but I suppose testing the device features works like this. The [deprecated method](https://developer.apple.com/documentation/metal/mtldevice/3081738-arebarycentriccoordssupported) returns `true`. 

```
        bool supportsShaderBarycentricCoordinates = false;
        if (@available(macOS 10.15, *)) {
            supportsShaderBarycentricCoordinates = _device.areBarycentricCoordsSupported;
        }
```

On the other hand the new method returns `false`.
```
        bool supportsShaderBarycentricCoordinates = false;
        if (@available(macOS 10.15, *)) {
            supportsShaderBarycentricCoordinates = _device.supportsShaderBarycentricCoordinates;
        }
```

**Comment by saschasc** [comment]
Out of curiosity, I tested the test program and also the Vulkan SDK on a Mac with almost the same configuration which is already running on macOS Ventura. There it works as expected. See attached vulkaninfo.

[vulkaninfo 1.3.236@ macOS 13.1 (22C65).txt](https://github.com/KhronosGroup/MoltenVK/files/10527704/vulkaninfo.1.3.236%40.macOS.13.1.22C65.txt)

Not sure how we could fix this. Probably additionally query the deprecated instance property?

**Comment by billhollings** [comment]
Thanks for your diligent detective work! Very helpful.

As you deduced, after the `1.3.216` release, we added a test to check whether the GPU actually supports barycentric coordinates. 

We are currently only using the `supportsShaderBarycentricCoordinates` Metal query to test for this. Your results in testing that vs `areBarycentricCoordsSupported` are interesting. Might be an inconsistency bug on some earlier Metal drivers, particularly since you indicate Ventura works as expected. 

And yes...probably doing a boolean OR of the current and deprecated Metal properties is probably the simplest fix for this. Should be fixed quickly.

**Comment by saschasc** [comment]
> Thanks for your diligent detective work! Very helpful.

Haha ... years of experience ;-). I'm looking forward to your presentation at the Vulkanised :-).

As long as the deprecated instance property is still available I think we can live with a boolean OR.

**Comment by billhollings** [comment]
This is fixed in PR #1832. Please retest and close this issue if fixed.

**Comment by saschasc** [comment]
It seems to be a little more elaborate. Also, I can't change my current development system that easily. However, I will be very happy to test the next VulkanSDK when an installer is available. Also, feel free to remind me should I not have closed the fix by then.
